rails/app/javascript/components/Map.jsx

```diff
@@ -28,18 +28,30 @@ export default class Map extends Component {
   };
 
   componentDidMount() {
-    this.map = new mapboxgl.Map({
-      container: this.mapContainer,
-      style: this.props.mapboxStyle,
-      center: [this.props.centerLong, this.props.centerLat],
-      zoom: this.props.zoom,
-      maxBounds: [
-        [this.props.sw_boundary_long, this.props.sw_boundary_lat], //southwest
-        [this.props.ne_boundary_long, this.props.ne_boundary_lat] //northeast
-      ],
-      pitch: this.props.pitch,
-      bearing: this.props.bearing
-    });
+    if (this.props.sw_boundary_long == null) { // no bounding box
+      this.map = new mapboxgl.Map({
+        container: this.mapContainer,
+        style: this.props.mapboxStyle,
+        center: [this.props.centerLong, this.props.centerLat],
+        zoom: this.props.zoom,
+        maxBounds: null, // no bounding box
+        pitch: this.props.pitch,
+        bearing: this.props.bearing
+      });
+    } else { // map does have bounding box
+      this.map = new mapboxgl.Map({
+        container: this.mapContainer,
+        style: this.props.mapboxStyle,
+        center: [this.props.centerLong, this.props.centerLat],
+        zoom: this.props.zoom,
+        maxBounds: [
+          [this.props.sw_boundary_long, this.props.sw_boundary_lat], //southwest
+          [this.props.ne_boundary_long, this.props.ne_boundary_lat] //northeast
+        ],
+        pitch: this.props.pitch,
+        bearing: this.props.bearing
+      });
+    }
```

Comment:
I might suggest simplifying this code and `resetMapToCenter()` somewhat. Instead of repeating the map definition, you could have some code before like:

```
    let mapBounds = null; 
    if (this.props.sw_boundary_long != null) {
      mapBounds = [
      [this.props.sw_boundary_long, this.props.sw_boundary_lat], //southwest
      [this.props.ne_boundary_long, this.props.ne_boundary_lat] //northeast
      ]
    }
```

And then set `maxBounds` to `mapBounds`.

Or maybe using that `boundary` definition?

Also, here we might want to check for all of the boundary lat/longs, not just `sw_boundary_long`.

Reply:
I just pushed a new commit simplifying the code and checking for all of the boundary lat/longs! Thanks for the feedback.